execution/gethexec/wasmstorerebuilder.go

```
@@ -0,0 +1,115 @@
+var RebuildingStartBlockHashKey []byte = []byte("_rebuildingStartBlockHash") // contains the block hash of starting block when rebuilding of wasm store first began
+var RebuildingDone common.Hash = common.BytesToHash([]byte("_done"))         // indicates that the rebuilding is done, if RebuildingPositionKey holds this value it implies rebuilding was completed
+
+func ReadFromKeyValueStore[T any](store ethdb.KeyValueStore, key []byte) (T, error) {
+	var empty T
+	posBytes, err := store.Get(key)
+	if err != nil {
+		return empty, err
+	}
+	var val T
+	err = rlp.DecodeBytes(posBytes, &val)
+	if err != nil {
+		return empty, fmt.Errorf("error decoding value stored for key in the KeyValueStore: %w", err)
+	}
+	return val, nil
+}
+
+func WriteToKeyValueStore[T any](store ethdb.KeyValueStore, key []byte, val T) error {
+	valBytes, err := rlp.EncodeToBytes(val)
+	if err != nil {
+		return err
+	}
+	err = store.Put(key, valBytes)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// RebuildWasmStore function runs a loop looking at every codehash in diskDb, checking if its an activated stylus contract and
+// saving it to wasm store if it doesnt already exists. When errored it logs them and silently returns
+//
+// It stores the status of rebuilding to wasm store by updating the codehash (of the latest sucessfully checked contract) in
+// RebuildingPositionKey every 50 checks.
+//
+// It also stores a special value that is only set once when rebuilding commenced in RebuildingStartBlockHashKey as the block
+// time of the latest block when rebuilding was first called, this is used to avoid recomputing of assembly and module of
+// contracts that were created after rebuilding commenced since they would anyway already be added during sync.
+func RebuildWasmStore(ctx context.Context, wasmStore ethdb.KeyValueStore, l2Blockchain *core.BlockChain, position, rebuildingStartBlockHash common.Hash) {
```

have that return an error instead of the log.Errors, and have the calling function check if err != nil